django/contrib/auth/views.py

```diff
@@ -122,9 +124,18 @@ def dispatch(self, request, *args, **kwargs):
             return HttpResponseRedirect(next_page)
         return super().dispatch(request, *args, **kwargs)
 
+    def get(self, request, *args, **kwargs):
+        # Deprecated in Django 3.1.
```

Comment:
I don't think we use such comments - the warning and docs explain enough

Reply:
How would the code in 4.0 look then? Just removing `get` will not do the trick since `TemplateView` defines it. IMO we should write the code as it would behave in 4.0 and then add the deprecation to the `get` part.

Reply:
`return HttpResponseNotAllowed()` should be all that's required.

Reply:
That would be counter-intuitive, not implementing `get` in the first place would already give you the wanted behaviour. Changing from `TemplateView` to `TemplateResponseMixin, ContextMixin, View` would get rid of the method.

Reply:
My idea was the same as Adam's: return `405` with a response body like "Logging out with GET is not allowed, please use POST if you want to log out". I am also fine with changing the base classes though.

Reply:
The `View` class does this by default for unimplemented methods ;) Hence I'd like a solution where we can just drop the `get` in 4.0.

Reply:
Yes that's fair, I haven't used CBV's so much recently and forget there are mixins for the Template and Context behaviour.